File: algo/5.kt
import kotlin.collections.maxBy

class Solution {
    fun find(s: String, left: Int, right: Int): String {
        var l = left
        var r = right
        while (l >= 0 && r <= s.length && s[l] == s[r - 1]) {
            l--
            r++
        }
        return s.substring(l + 1, r - 1)
    }

    fun longestPalindrome(s: String): String {
        var result = ""
        for (i in 0 until s.length) {
            result = arrayOf(
                result,
                find(s, i, i + 1),
                find(s, i, i + 2)
            ).maxBy { it.length } ?: "" // 코틀린 버전 -_-
        }
        return result
    }
}

Solution().longestPalindrome("cbbd")

File: algo/5.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if len(s) < 2 or s == s[::-1]:
            return s
        
        def expand(left: int, right: int) -> str:
            while left >= 0 and right <= len(s) and s[left] == s[right - 1]:
                left -= 1
                right += 1
            return s[left + 1:right - 1]
        
        result = ''

        for i in range(len(s) - 1):
            result = max(
                result,
                expand(i, i + 1),
                expand(i, i + 2),
                key=len
            )
        return result
    
s = "babad"
a = Solution()
print(a.longestPalindrome(s))

s = "cbbd"
print(a.longestPalindrome(s))